torchreid/integration/nncf/compression.py

```diff
@@ -0,0 +1,247 @@
+                                                               # will be a closest multiple of the batch size.
+                            #"num_bn_forget_samples": 1024, # Number of samples from the training
+                                                            # dataset to pass through the model at initialization in order to erase
+                                                            # batchnorm statistics of the original model (using large momentum value
+                                                            # for rolling mean updates). The actual number of samples will be a
+                                                            # closest multiple of the batch size.
+                            }
+                        }
+                    }
+                ],
+            "log_dir": "."
+        }
+
+    # do it even if nncf_config_data is loaded from a checkpoint -- for the rare case when
+    # the width and height of the model's input was changed in the config
+    # and then finetuning of NNCF model is run
+    h, w = cfg.data.height, cfg.data.width
+    nncf_config_data.setdefault('input_info', {})
+    nncf_config_data['input_info']['sample_size'] = [1, 3, h, w]
+
+    nncf_config = NNCFConfig(nncf_config_data)
+    print(f'nncf_config =\n{pformat(nncf_config)}')
+    if not nncf_metainfo:
+        nncf_metainfo = create_nncf_metainfo(nncf_compression_enabled=True,
+                                             nncf_config=nncf_config_data)
+    else:
+        # update it just to be on the safe side
+        nncf_metainfo['nncf_config'] = nncf_config_data
+
+    class ReidInitializeDataLoader(InitializingDataLoader): #TODO: check is it correct
+        def get_inputs(self, dataloader_output):
+            # define own InitializingDataLoader class using approach like
+            # parse_data_for_train and parse_data_for_eval in the class Engine
+            # dataloader_output[0] should be image here
+            args = (dataloader_output[0], )
+            return args, {}
+
+    cur_device = next(model.parameters()).device
+    print(f'NNCF: cur_device = {cur_device}')
+
+    if checkpoint_path is None:
+        print('No NNCF checkpoint is provided -- register initialize data loader')
+        train_loader = datamanager_for_init.train_loader
+        wrapped_loader = ReidInitializeDataLoader(train_loader)
+        nncf_config = register_default_init_args(nncf_config, wrapped_loader, device=cur_device)
+        resuming_state_dict = None
+    else:
+        print(f'Loading NNCF model from {checkpoint_path}')
+        resuming_state_dict = _load_checkpoint_for_nncf(model, checkpoint_path, map_location=cur_device)
+        print(f'Loaded NNCF model from {checkpoint_path}')
+
+    transform = build_inference_transform(
+        cfg.data.height,
+        cfg.data.width,
+        norm_mean=cfg.data.norm_mean,
+        norm_std=cfg.data.norm_std,
+    )
+    def random_image(height, width):
```

Comment:
`random_image` was already defined in `scripts.script_utils`

Reply:
Ok

Reply:
Fixed.
Please, note that to use `random_image` in `torchreid.integration.nncf.compression` I had to move it to `torchreid.utils.tools`, otherwise cyclic import issue happened.